Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -3184,6 +3220,27 @@ namespace AssetProcessor
                 // Do not process this file yet.  When the job is done it will retrigger processing for this file.
                 return;
             }
+
+            AZ::Outcome<QString, void> topmostSourceAsset = GetFirstTopmostSourceAssetForProductPath(productName);
+
+            if (topmostSourceAsset.IsSuccess())
+            {
+                if (!QFile::exists(topmostSourceAsset.GetValue()))
+                {
+                    AZ_Printf("AP", "Topmost source '%s' for intermediate asset '%.*s' no longer exists, "
+                        "so this intermediate asset will not be reprocessed.",
+                        topmostSourceAsset.GetValue().toUtf8().data(),
+                        AZ_STRING_ARG(productName));
+                    return;
```

Comment:
I think this should be calling `CheckDeletedSourceFile` to clean up the asset.  Or does that happen later during CheckMissingFiles?

Reply:
![image](https://user-images.githubusercontent.com/4838196/217928165-1938a687-7a41-4229-8242-4bc308c83fac.png)

Here's an example from one of the tests, at least, that shows the new message followed by the delete. This is a contrived situation, the test specifically calls these, though, but I think it shows the general idea and that cleanup does happen when AssessDeletedFile is eventually called on the source.